File: Languages/Ruby/bee_movie.rb
# Hello, World!
puts "Hello, World!"

def analyze_bee_movie
  script_path = "../beemoviescript.txt"
  
  unless File.exist?(script_path)
    puts "File not found: #{script_path}"
    return
  end
  
  puts "Bee Movie Script:"
  puts "--------------------------------------------------"
  
  letter_counts = Hash.new(0)
  total_letters = 0
  
  # Read file line by line and print each line
  File.readlines(script_path).each do |line|
    puts line
    
    # Count letters
    line.each_char do |char|
      if char.match?(/[a-zA-Z]/)
        lower = char.downcase
        letter_counts[lower] += 1
        total_letters += 1
      end
    end
  end
  
  puts "--------------------------------------------------"
  puts "Analysis complete."
  
  if total_letters > 0
    # Sort by count (descending)
    sorted = letter_counts.sort_by { |letter, count| -count }
    
    puts "\nTop 3 most commonly used letters:"
    sorted.take(3).each_with_index do |(letter, count), index|
      puts "#{index + 1}. '#{letter}': #{count} times"
    end
  else
    puts "No letters found in the script."
  end
end

analyze_bee_movie

File: Languages/Elixir/bee_movie.exs
# Hello, World!
IO.puts "Hello, World!"

def analyze_bee_movie do
  script_path = "../beemoviescript.txt"
  
  unless File.exists?(script_path) do
    IO.puts "File not found: #{script_path}"
    return
  end
  
  IO.puts "Bee Movie Script:"
  IO.puts "--------------------------------------------------"
  
  # Read file line by line and print each line
  content = File.read!(script_path)
  lines = String.split(content, "\n")
  
  # Count letters
  letter_counts = 
    content
    |> String.graphemes()
    |> Enum.reduce(%{}, fn char, acc ->
      if String.match?(char, ~r/[a-zA-Z]/) do
        lower = String.downcase(char)
        Map.update(acc, lower, 1, &(&1 + 1))
      else
        acc
      end
    end)
  
  # Print each line
  Enum.each(lines, &IO.puts/1)
  
  IO.puts "--------------------------------------------------"
  IO.puts "Analysis complete."
  
  if map_size(letter_counts) > 0 do
    # Sort by count (descending) and take top 3
    top_3 = 
      letter_counts
      |> Enum.sort_by(fn {_letter, count} -> -count end)
      |> Enum.take(3)
    
    IO.puts "\nTop 3 most commonly used letters:"
    Enum.with_index(top_3, 1)
    |> Enum.each(fn {index, {letter, count}} ->
      IO.puts "#{index}. '#{letter}': #{count} times"
    end)
  else
    IO.puts "No letters found in the script."
  end
end

analyze_bee_movie()
